### PR TITLE
Add components for Supporter Plus Only checkout test

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This adds the following new components:

- A simplified price cards component limited to two options, with an accordion section for product details
- An extracted checklist component from the existing checkout with a new 'compact' variation
- An 'info box' component to display price and renewal information

[**Trello Card**](https://trello.com/c/xa882tGK)

## Why are you doing this?

These components are for use in an upcoming A/B test.

## Accessibility test checklist
 - [x] [Tested with screen reader](https://webaim.org/articles/voiceover/)
 - [x] [Navigable with keyboard](https://www.accessibility-developer-guide.com/knowledge/keyboard-only/browsing-websites/)
 - [x] [Colour contrast passed](https://www.whocanuse.com/)

## Screenshots

Screenshot 2023-06-21 at 13 27 44

Screenshot 2023-06-21 at 13 27 34

### CODE REVIEW

**Comment by i-hardy** [comment]
[S+ only Landing Page part 1: Create S+ Only Pricing Card Component](https://trello.com/c/xa882tGK/1354-s-only-landing-page-part-1-create-s-only-pricing-card-component)

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListData.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import type { SerializedStyles } from '@emotion/react';
 import { css } from '@emotion/react';
-import { neutral } from '@guardian/source-foundations';
+import { palette } from '@guardian/source-foundations';
```
Comment (marialani) [inline support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListData.tsx:3]: Just a general question, do we prefer to use the palette going forward?

File: support-frontend/assets/components/priceCards/simplePriceCards.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	height: min-content;
+	justify-content: center;
+	text-decoration: none;
+	${textSans.xsmall()};
+	color: ${palette.neutral[20]};
+`;
+
+const iconCss = (flip: boolean) => css`
+	svg {
+		max-width: 12px;
```
Comment (marialani) [inline support-frontend/assets/components/priceCards/simplePriceCards.tsx:80]:
```suggestion
		max-width: ${space[3]}px;
```

**Review by marialani — APPROVED** [review]
Almost splitting hairs here, this all looks good to me!

File: support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListData.tsx (hunk 1 above)
Comment (i-hardy) [reply support-frontend/assets/components/checkoutBenefits/checkoutBenefitsListData.tsx:3]: Yes, the direct use of `neutral`, `brand` etc is deprecated in the most recent version of Source so we should be using `palette` (this should get flagged in VSCode so it's hard to miss).

**Pull request merged**